### DISCUSSION TITLE
iMAT+MTA functions on Python

### DISCUSSION THREAD

**Discussion by RolandLendsley** [opencobra/cobrapy#1133] [General]
is the iMAT (https://github.com/opencobra/cobratoolbox/blob/6c1ba69dfd7adbf121a3b987a3446b3d7fdc15ed/src/dataIntegration/transcriptomics/iMAT/iMAT.m) and MTA algorithm from cobratoolbox(matlab) available on cobrapy too?

If not, would you think about implementing it soon on cobrapy?
If that's not the case, could you provide a swift alternative to run iMAT/MAT within a Python based project?

Thank you,
Roland

**Comment by cdiener** [comment]
Most of the metabolic reconstruction algorithms were slated to be added to [driven](https://github.com/opencobra/driven). However, that has been stalled since nobody wanted to pick this up. There is an implementation of iMAT [in there](https://github.com/opencobra/driven/blob/c0f7f5e391362bd87cbc001b6fadf5960c9a655b/driven/flux_analysis/transcriptomics.py#L103) but I would not bet that it still works (last updated several years ago). There are sone actively maintained alternatives on https://opencobra.github.io/cobrapy/tags/reconstruction/ you could check out.

**Comment by RolandLendsley** [comment]
Dear Christian, thank you very much,

I found iMAT but no MTA or rMTA in Cobrapy, am I missing something out here?

Since they should be used in conjuction in my case, then I would be still directed into using matlab because one of those two runs on matlab only

Let me know, thank you,
Roland

**Comment by cdiener** [comment]
Not sure what exactly you mean by MTA since there are several algorithms with that acronym. If you mean the metabolic transformation algorithm, no, there is no implementation in cobrapy I am aware of :cry: Though it should not be too hard to write a package that implements them if you would like to take a stab at it. Optlang helps quite a bit.

**Comment by y1zhou** [comment]
Late to the discussion, but it seems several more algorithms were implemented in the [driven devel branch](https://github.com/opencobra/driven/tree/devel/src/driven/omics).
